python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -62,19 +66,45 @@ _DECIMAL_AGGS = {"COUNT", "SUM", "ARGMIN", "ARGMAX", "MIN", "MAX", "NUNIQUE",
 # workaround for https://github.com/cython/cython/issues/3885
 ctypedef const scalar constscalar
 
+
+cdef _agg_result_from_columns(
+    vector[libcudf_groupby.aggregation_result]& c_result_columns,
+    set column_orders,
+    int n_input_columns
+):
+    """Construct the list of result columns from libcudf result. The result
+    contains the same number of lists as the number of input columns. Result
+    for an input column that has no applicable aggregations is an empty list.
+    """
+    cdef int n_res_cols = c_result_columns.size()
+    result_columns = []
+    cdef int i, j, result_index = 0
+    cdef vector[unique_ptr[column]]* c_results_i
+    for i in range(n_input_columns):
+        if i in column_orders:
+            c_results_i = &c_result_columns[result_index].results
```

Comment:
Nit: with the new indexing scheme, probably better to name this just `c_result` or something. It no longer corresponds to index `i`, but rather the `result_index`.